doc/runtimes/index.rst

```diff
@@ -0,0 +1,97 @@
+computing platforms and operating systems.
+
+The :doc:`/commands/build`, :doc:`/commands/view`, and :doc:`/commands/shell`
+commands all require a runtime, as they require access to other Nextstrain
+software.
+
+The :doc:`/commands/setup`, :doc:`/commands/check-setup`, and
+:doc:`/commands/update` commands manage the runtimes available for use by the
+commands above.  The :doc:`/commands/version` command's :option:`--verbose
+<nextstrain version --verbose>` option reports detailed version information
+about all available runtimes.
+
+Other Nextstrain CLI commands, such as the :doc:`/commands/remote/index` family
+of commands and the related :doc:`/commands/login` and :doc:`/commands/logout`
+commands, do not require a runtime.  They may be used without any prior set up
+of a runtime.
+
+The runtimes currently available are the:
+
+  - :doc:`/runtimes/docker`
+  - :doc:`/runtimes/singularity`
+  - :doc:`/runtimes/conda`
+  - :doc:`/runtimes/ambient`
+  - :doc:`/runtimes/aws-batch`
+
+Runtimes are managed (maintained, tested, versioned, released) by the
+Nextstrain team, except for the ambient runtime.  The ambient runtime is
+special in that it's whatever computing environment in which Nextstrain CLI
+itself is running (that is, it's managed by the user).
+
+You can set up and use multiple runtimes on the same computer, for example if
+you want to use them in different contexts.  Runtime-using commands let you
+select a different runtime than your default with command-line options (e.g.
+``--docker``, ``--conda``, and so on).  For example, you might use the Docker
+runtime to run small builds locally and then use the AWS Batch runtime to run
+large scale builds with more computing power.
+
+If you pick one runtime and later realize you want to switch, you can go back
+and set up the other.
+
+
+Comparison
+==========
+
+.. csv-table::
+    :file: comparison.csv
+    :header-rows: 1
+    :stub-columns: 1
+
+Isolation level
+    A relative ranking from least isolated (*none*, 0) to most isolated
+    (*great*, 3) from the underlying computer system.
+
+Containerized?
+    A containerized :term:`computing platform` provides a higher degree of
+    isolation, which in turn usually means a higher degree of portabililty and
+    reproducibility across different computers and users.
+
+Locality
+    *Local* means "on the same computer where ``nextstrain`` is being run".
+    *Remote* means "on a different computer".
+
+    Docker is most often used to run containers locally, but can also be used
+    to run them remotely.
+
+External dependencies
+    Third-party programs or configuration which are required to use a runtime
+    but not managed by :doc:`/commands/setup` and :doc:`/commands/update`.
+
+
+Compatibility
+=============
+
+Switching runtimes or updating the version of a runtime may result in different
+versions of Nextstrain components like Augur and Auspice as well as other
+programs, and thus different behaviour.
```

Comment:
I found myself asking [2] [recently](https://discussion.nextstrain.org/t/pandas-error-during-augur-filter-nonetype-object-has-no-attribute-str/1420/4?u=victorlin) when Augur 22.2.0 had a bug. It took me some digging in the Docker image's build logs to figure out that `nextstrain/base:build-20230720T001758Z` was the last runtime version with an earlier version of Augur.

I don't think there's an easy solution here.

One solution would be to maintain an up-to-date resource describing the versions of Nextstrain programs in each runtime version. For example:

| runtime | version                                | Augur  | Auspice | Nextstrain CLI |
|---------|----------------------------------------|--------|---------|----------------|
| conda   | nextstrain-base 20230815T151610Z       | 22.3.0 | 2.46.0  | 7.1.0          |
| conda   | nextstrain-base 20230805T043657Z       | 22.2.0 | 2.46.0  | 7.1.0          |
| docker  | nextstrain/base:build-20230814T140428Z | 22.3.0 | 2.46.0  | 7.1.0          |
| docker  | nextstrain/base:build-20230805T043648Z | 22.2.0 | 2.46.0  | 7.1.0          |
| …       | …                                      | …      | …       | …              |

This would be straightforward to use, but probably not straightforward to maintain.

Reply:
Pushed 8b90a0b to address 1. Addressing 2 seems useful, I agree, but I think to provide a useful answer besides "binary search it yourself" we'll need to produce static reports for each runtime version of the included software (ours and others). This would also address your broader comment above.